src/lib/components/submit/SubmitAddNodeProvider.svelte

```diff
@@ -1,9 +1,57 @@
 <script lang="ts">
 	import Input from '$lib/components/ui/Input.svelte';
+	import { setMetadata } from '$lib/services/idb.services';
+	import { debounce } from '@dfinity/utils';
+	import type { ProposalEditableMetadata } from '$lib/types/juno';
+
+	export let metadata: ProposalEditableMetadata | undefined;
+
+	// TODO: make the fields more fine grained and then construct the proposal according to a template behind the scenes
 
 	let title = '';
 	let summary = '';
-	let nodeProviderPid = '';
+	let nodeProviderId = '';
+
+	const init = () => {
```

Comment:
We need to refactor the code in another PR, it's basically a copy/paste.
We can notably probably extract a component for the title and the url and probably find a way to making saving data generic.

Can you add a TODO?